packages/lambda-api-tracker/src/__test__/xyz-request.test.ts

```diff
@@ -0,0 +1,81 @@
+// process.env['COG_BUCKET'] = 'fake-bucket';
+import { LambdaHttpResponseCloudFront, LambdaSession, LogConfig, LogType } from '@basemaps/lambda-shared';
+import { CloudFrontRequestEvent } from 'aws-lambda';
+import * as o from 'ospec';
+import { LogFn } from 'pino';
+import { handleRequest } from '../index';
+import { ValidateRequest } from '../validate';
+
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+o.spec('xyz-request', () => {
+    const origValidate = ValidateRequest.validate;
+    let mockLogger: LogType;
+
+    /** Generate mock CloudFrontRequestEvent */
+    function req(uri = '', querystring = '?api=12345'): CloudFrontRequestEvent {
+        return {
+            Records: [
+                {
+                    cf: {
+                        request: {
+                            method: 'get',
+                            querystring,
+                            uri,
+                            headers: {
+                                referer: ['from/url'],
+                                'user-agent': ['test browser'],
+                            },
+                            clientIp: '1.2.3.4',
+                        },
+                    },
+                },
+            ],
+        } as any;
+    }
+
+    o.beforeEach(() => {
+        mockLogger = LogConfig.get();
```

Comment:
you can disable the logger with `LogConfig.disable()`

by overwriting with the spy you are destroying the logger forever.